src/screens/Drafting/Draft.tsx

```diff
@@ -1,26 +1,238 @@
 import React, { ReactElement } from 'react';
-import { StyleSheet, Text, View } from 'react-native';
+import { Text, View, Pressable, Switch } from 'react-native';
+import { useAuthentication } from '../../utils/hooks/useAuthentication';
 import { Button } from 'react-native-elements';
+import { getAuth, signOut } from 'firebase/auth';
+import { styles } from './styles';
+import { Job } from '../../types/types';
+import { createJob } from '../../firebase/firestore/job';
+import { ScrollView } from 'react-native-gesture-handler';
+import FormInput from '../../components/JobPostFormInput/JobPostFormInput';
+import { FormProvider, SubmitHandler, useForm } from 'react-hook-form';
+
+const auth = getAuth();
 
 const DraftScreen = ({ navigation }: any): ReactElement => {
+  const { user } = useAuthentication();
+
+  const [dateIsEnabled, setDateIsEnabled] = React.useState(false);
+  const [companyNameIsEnabled, setCompanyNameIsEnabled] = React.useState(false);
+  const [addressIsEnabled, setAddressIsEnabled] = React.useState(false);
+  const [contactPersonIsEnabled, setContactPersonIsEnabled] = React.useState(false);
+  const [phoneIsEnabled, setPhoneIsEnabled] = React.useState(false);
+  const [jobPositionIsEnabled, setJobPositionIsEnabled] = React.useState(false);
+  const [languageReqIsEnabled, setLangaugeReqIsEnabled] = React.useState(false);
+  const [workingHoursIsEnabled, setWorkingHoursIsEnabled] = React.useState(false);
+  const [workingDaysIsEnabled, setWorkingDaysIsEnabled] = React.useState(false);
+  const [salaryIsEnabled, setSalaryIsEnabled] = React.useState(false);
+  const [probationPeriodIsEnabled, setProbationPeriodIsEnabled] = React.useState(false);
+  const [employeeBenefitIsEnabled, setEmployeeBenefitIsEnabled] = React.useState(false);
+  const [otherInfoIsEnabled, setOtherInfoIsEnabled] = React.useState(false);
+
+  interface FormValues {
+    date: string;
+    companyName: string;
+    address: string;
+    contactPerson: string;
+    phone: string;
+    jobPosition: string;
+    languageRequirement: string;
+    workingHours: string;
+    workingDays: string;
+    salary: string;
+    probationPeriod: string;
+    employeeBenefit: string;
+    otherInfo: string;
+  }
+  const { ...methods } = useForm();
+
+  const onSubmit: SubmitHandler<FormValues> = async (data) => {
```

Comment:
we should add validation to ensure that the form won't submit without an input for date — this can be done in another PR